xwiki-pro-macros-ui/src/main/resources/Confluence/Macros/ViewFile.xml

```diff
@@ -256,14 +256,16 @@ require(['jquery', 'xwiki-meta', 'xwiki-suggestAttachments'], function($, xm) {
     }
 
     const decorateWithCollaboraButton = function(viewFile, fileRef, spanContainer) {
-      const fileParts = fileRef.split('@');
+      const attachment = XWiki.Model.resolve(fileRef, XWiki.EntityType.ATTACHMENT);
+      var fileName = attachment.getName();
+      var fileParent = attachment.parent.toString();
```

Comment:
This is a document reference? If yes, then you shouldn't use toString, but a string serializer